test/gtest/common/exclusive_scan.cpp

```diff
@@ -0,0 +1,134 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+
+// Fixture
+template <typename T> class ExclusiveScan : public testing::Test {
+public:
+};
+
+TYPED_TEST_SUITE(ExclusiveScan, AllTypes);
+
+TYPED_TEST(ExclusiveScan, whole_range) {
+  TypeParam dv_in(5);
+  xhp::iota(dv_in, 1);
+  TypeParam dv_out(5, 0);
+
+  xhp::exclusive_scan(dv_in, dv_out, 10, std::plus<>());
+  EXPECT_EQ(10, dv_out[0]);
+  EXPECT_EQ(10 + 1, dv_out[1]);
+  EXPECT_EQ(10 + 1 + 2, dv_out[2]);
+  EXPECT_EQ(10 + 1 + 2 + 3, dv_out[3]);
+  EXPECT_EQ(10 + 1 + 2 + 3 + 4, dv_out[4]);
+}
+
+TYPED_TEST(ExclusiveScan, empty) {
+  TypeParam dv_in(3, 1);
+  TypeParam dv_out(3, 0);
+  xhp::exclusive_scan(rng::begin(dv_in), rng::begin(dv_in), rng::begin(dv_out),
+                      0);
+  EXPECT_EQ(0, dv_out[0]);
+  EXPECT_EQ(0, dv_out[1]);
+  EXPECT_EQ(0, dv_out[2]);
```

Comment:
The above conditions can be replaced with one, giving even more information about the failure (all differences in single message):
`EXPECT_TRUE(equal({0, 0, 0}, dv_out));`
The same in tests below.

Reply:
I prefer my version, it is much clearer, thanks ;)